lib/share_progress/button_client.rb

```diff
@@ -0,0 +1,40 @@
+require 'share_progress'
+
+module ShareProgress
+  class ButtonClient
+    def initialize
+      @connection = ShareProgress::ConnectionFactory.get_connection
+    end
+
+    def new
+    end
+
+    def update
+    end
+
+    def find(id)
+      @connection.make_call endpoint('read'), 'get', { id: id }
+    end
+
+    def destroy(id)
+      @connection.make_call endpoint('delete'), 'post', { id: id }
+    end
+
+    def all(options={})
+      acceptable = [:offset, :limit]
+      @connection.make_call endpoint, 'get', filter_keys(options, acceptable)
+    end
```

Comment:
The reason that I'd gone with a singular connection class that just exposed a wrapper for HTTParty was in case we were ever in a position where HTTParty stopped being the correct choice for HTTP connections in Ruby; it would make refactoring to support whatever the new library was quick and easy, because HTTParty would only surface in one part of the code base.